### PR TITLE
fix: call language client stop during deactivation

### PR DESCRIPTION
**Pull request by peternhale**
This PR fixes the issue of language servers being orphaned. 

This change adds a call to the language client stop function during the apex extension deactivate call.

@W-14446753@

### CODE REVIEW

File: packages/salesforcedx-vscode-apex/src/index.ts [hunk 1]
```diff
@@ -284,6 +284,7 @@ async function registerTestView(): Promise<vscode.Disposable> {
 }
 
 export async function deactivate() {
+  await languageClient?.stop();
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex/src/index.ts:287]: sweet.  We already have a unit test suite for this class.  Mind adding a unit test for the deactivate function?  I'll QE in the mean time.

**Review by gbockus-sf — APPROVED** [review]
Thanks for the quick fix!
QE passed. 

Test Cases

* Verify when run via a vsix the Apex LSP exits when VSCode is shut down via the X ✅
* Verify when run via a vsix the Apex LSP exits when VSCode is shut down via quit Visual Studio Code ✅
* Verify when run via a vsix the Apex LSP exits when VSCode is shut down when running in dev mode ✅

File: packages/salesforcedx-vscode-apex/src/index.ts (hunk 1 above)
Comment (peternhale) [reply packages/salesforcedx-vscode-apex/src/index.ts:287]: Will do.

File: packages/salesforcedx-vscode-apex/src/index.ts [hunk 2]
```diff
@@ -284,20 +284,25 @@ async function registerTestView(): Promise<vscode.Disposable> {
 }
 
 export async function deactivate() {
-  await languageClient?.stop();
+  await getClient()?.stop();
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex/src/index.ts:287]: Why not just `languageClientUtils.getClientInstance().stop()`?

File: packages/salesforcedx-vscode-apex/test/jest/index.test.ts [hunk 3]
```diff
@@ -62,3 +69,23 @@ describe('indexDoneHandler', () => {
     expect(apexLSPStatusBarItemMock).toHaveBeenCalledTimes(1);
   });
 });
+
+class MockApexLanguageClient {
+  constructor() { }
+  start() { }
+  stop() { }
+}
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex/test/jest/index.test.ts:77]: I'm not sure why this is necessary? 

Would 
```
stopSpy = jest.fn();
jest.spyOn(languageClientUtils, 'getClientInstance').mockReturnValue({stop: stopSpy}); 
```
achieve the same thing without the extra class here and exported function in index?

**Review by gbockus-sf — APPROVED** [review]
one q about the test. Good to go.

**Review by gbockus-sf — APPROVED** [review]
🚢

**Pull request merged**